### PR TITLE
Resolve uid considering monikers

### PR DESCRIPTION
**Pull request by mingwli**
[AB#286912](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/286912)
[AB#290850](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/290850)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6458)

### CODE REVIEW

**Comment by mingwli** [comment]
May need more spec test :)

File: src/docfx/build/xref/XrefResolver.cs [hunk 1]
```diff
@@ -217,8 +215,23 @@ private static string RemoveSharingHost(string url, string hostName)
...
-            if (_internalXrefMap.Value.TryGetValue(uid, out var spec))
+            if (_internalXrefMap.Value.TryGetValue(uid, out var specs))
             {
+                var spec = default(InternalXrefSpec);
+                if (specs.Length == 1)
+                {
+                    spec = specs[0];
+                }
+
+                var referencingMonikers = _monikerProvider.GetFileLevelMonikers(ErrorWriter.Null, referencingFile.FilePath);
```
Comment (mingwli) [inline src/docfx/build/xref/XrefResolver.cs:226]: should be in a else clause

**Comment by herohua** [comment]
Yes, especially on the negative case of intersection

---
In reply to: [680764595](https://github.com/dotnet/docfx/pull/6458#issuecomment-680764595) [](ancestors = 680764595)

File: src/docfx/build/xref/XrefResolver.cs [hunk 2]
```diff
@@ -217,8 +215,26 @@ private static string RemoveSharingHost(string url, string hostName)
...
+            if (_internalXrefMap.Value.TryGetValue(uid, out var specs))
             {
+                var spec = default(InternalXrefSpec);
+                if (specs.Length == 1)
+                {
+                    spec = specs[0];
+                }
+                else
+                {
+                    var referencingMonikers = _monikerProvider.GetFileLevelMonikers(ErrorWriter.Null, referencingFile.FilePath);
```
Comment (live1206) [inline src/docfx/build/xref/XrefResolver.cs:227]: nit: use `ErrorBuilder.Null`

File: src/docfx/build/xref/XrefResolver.cs [hunk 3]
```diff
@@ -137,21 +139,17 @@ public XrefMapModel ToXrefMapModel(bool isLocalizedBuild)
...
-                    // DHS appends branch information from cookie cache to URL, which is wrong for UID resolved URL
-                    // output xref map with URL appending "?branch=master" for master branch
-                    var query = repositoryBranch != "live" ? $"?branch={repositoryBranch}" : "";
-                    var href = UrlUtility.MergeUrl($"https://{_xrefHostName}{xref.Href}", query);
-
-                    var xrefSpec = xref.ToExternalXrefSpec(href);
-                    return xrefSpec;
-                })
-                .OrderBy(xref => xref.Uid).ToArray();
+                : (from xrefs in _internalXrefMap.Value.Values
```
Comment (live1206) [inline src/docfx/build/xref/XrefResolver.cs:147]: I feel the original expression is more readable...
Just simply change to `xrefList` and `xrefList.First().ToExternalXrefSpec`

File: src/docfx/build/xref/XrefResolver.cs [hunk 4]
```diff
@@ -217,8 +221,26 @@ private static string RemoveSharingHost(string url, string hostName)
 
         private (IXrefSpec?, string? href) ResolveInternalXrefSpec(string uid, Document referencingFile, Document inclusionRoot)
```
Comment (yufeih) [inline src/docfx/build/xref/XrefResolver.cs:222]: Consider adding an optional `MonikerList?` as a parameter, and switch behaviors based on whether the resolve carries over monikers.

**Review by yufeih — APPROVED** [review]

**Pull request merged**